smart-contracts/contracts/CosmosBridge.sol

```diff
@@ -179,35 +188,36 @@ contract CosmosBridge is CosmosBridgeStorage, Oracle {
         require(sigs.length == claims.length, "INV_CLM_LEN");
         require(sigs.length == signatureData.length, "INV_SIG_LEN");
 
+        uint256 intermediateNonce = lastNonceSubmitted;
+        require(intermediateNonce + 1 == claims[0].nonce, "INV_ORD");
+
```

Comment:
Move line 199 to line 193 and instead of assigning to intermediate nonce, just increment by claims.length. This would remove the need to have a reentrancy lock.